### PR TITLE
Adds Create Camel project to the welcome screen

### PR DESCRIPTION
**Pull request by hdamarcelo**
Adds Create Camel projecy to the welcome screen
    
- Rename NewCamelProjectCommand class to reflect its abstract nature.
- Create a 'generic' NewCamelProjectCommand that asks the user to select a runtime. Register the command with 'enablement = false' so that its does not show in the command palette.
- Refactor other classes with the new project structure.

welcome

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -194,17 +194,22 @@
         "category": "Camel",
         "enablement": "workspaceFolderCount != 0"
       },
+      {
+        "command": "camel.jbang.project.new",
+        "title": "Create a Camel project",
+        "category": "Camel",
+        "enablement": "false"
+
```
Comment (djelinek) [inline package.json:202]: empty line

File: src/commands/NewCamelQuarkusProjectCommand.ts [hunk 2]
```diff
@@ -16,13 +16,13 @@
...
-import { NewCamelProjectCommand } from "./NewCamelProjectCommand";
+import { AbstractNewCamelProjectCommand } from "./AbstractNewCamelProjectCommand";
 
-export class NewCamelQuarkusProjectCommand extends NewCamelProjectCommand {
+export class NewCamelQuarkusProjectCommand extends AbstractNewCamelProjectCommand {
 
 	public static readonly ID_COMMAND_CAMEL_QUARKUS_PROJECT = 'camel.jbang.project.quarkus.new';
 
-	getRuntime(): string {
+	async getRuntime(): Promise<string> {
```
Comment (djelinek) [inline src/commands/NewCamelQuarkusProjectCommand.ts:25]: is this really needs to be an `async` function?

File: src/commands/NewCamelSpringBootProjectCommand.ts [hunk 3]
```diff
@@ -16,13 +16,13 @@
...
-import { NewCamelProjectCommand } from "./NewCamelProjectCommand";
+import { AbstractNewCamelProjectCommand } from "./AbstractNewCamelProjectCommand";
 
-export class NewCamelSpringBootProjectCommand extends NewCamelProjectCommand {
+export class NewCamelSpringBootProjectCommand extends AbstractNewCamelProjectCommand {
 
 	public static readonly ID_COMMAND_CAMEL_SPRINGBOOT_PROJECT = 'camel.jbang.project.springboot.new';
 
-	getRuntime(): string {
+	async getRuntime(): Promise<string> {
```
Comment (djelinek) [inline src/commands/NewCamelSpringBootProjectCommand.ts:25]: is this really needs to be an `async` function?

File: src/ui-test/tests/commands.projects.test.ts [hunk 4]
```diff
@@ -37,7 +38,7 @@ import {
 } from '../utils/testUtils';
 import * as pjson from '../../../package.json';
 
-describe('Create a Camel Project using command', function () {
+describe.only('Create a Camel Project using command', function () {
```
Comment (djelinek) [inline src/ui-test/tests/commands.projects.test.ts:41]: remove `only` when code is ready

File: Changelog.md [hunk 5]
```diff
@@ -1,6 +1,7 @@
 # Change Log
 
 ## 1.6.0
+- Provide folder selection when using `Create Camel Quarkus/SpringBoot Project` command
```
Comment (djelinek) [inline Changelog.md:5]: empty line

**Review by apupier — CHANGES_REQUESTED** [review]
tests are failing on Windows

**Comment by hdamarcelo** [comment]
> tests are failing on Windows

After some testing using a VM I was able to reproduce the error we can see in the test pipeline but unfortnutatly not solve it (the error is the same in [PR1952](https://github.com/camel-tooling/camel-lsp-client-vscode/pull/1952))

Tests I made while trying to reproduce the error:

1. Execute `npm run ui-test` inside vscode integrated terminal: The issue is that after the command execution (after the task that runs `jbang camel export` completes) no pom.xml file is created and because of that the test assertion fails. During the test execution I can see the following error in the terminal among other warning messages (they do not appear on linux)
>ERR: [File Watcher (node.js)] Failed to watch c:\<path-to-project>\camel-lsp-client-vscode\src\ui-test\resources\create-camel-project-workspace\.vscode for changes using fs.watch() (Error: EPERM: operation not permitted, watch, undefined)", source: vscode-file://vscode-app/c:/<path-to-user-folder>/AppData/Local/Temp/test-resources/VSCode-win32-x64-archive/resources/app/out/vs/workbench/workbench.desktop.main.js (35)
  
2. Execute `npm run ui-test` outside vscode: Same result as 1
3. Execute `npm run ui-test` in windows bash: Same result as 1
4. Execute `npm run ui-test` in windows powershell: Same result as 1
5. Execute `npm run ui-test` changing camel.jbang.version to 4.8.0: Same result as 1
6. Running the command "Create a Camel Project" after running the extension (hitting F5): `jbang camel export` fails due to a simultaneous lock in a NTUSER.DAT file inside .camel-jabng folder (file is deleted after command execution)
7. Running the command "Create a Camel Project" after running the extension (hitting F5) as administrator: Same result as 6
8. Running the command "Create a Camel Project" after installing the built vsix (without running the project in another vscode instance): pom.xml file is created as expected

While adding pauses in each step of the ui-test execution it seems to me that the camel jbang process ends early and does not create the pom.xml file during the test suite execution.

I also tried executing camel jbang via a child process intead of using vscode's ShellExecution method but that also did not work.

**Review by apupier — CHANGES_REQUESTED** [review]
side-tracked and wasn't able to test it locally yet but would like to provide the initial feedback

File: src/commands/AbstractNewCamelProjectCommand.ts [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+				await this.copyFile(`../../../resources/${runtime}/${filename}.json`, path.join(outputFolder.fsPath, `.vscode/${filename}.json`));
+			}
+
+			// open the newly created project in a new vscode instance
+			await commands.executeCommand('vscode.openFolder', outputFolder, true);
+
+		}
+	}
+
+	abstract getRuntime(): Promise<string|undefined>;
```
Comment (apupier) [inline src/commands/AbstractNewCamelProjectCommand.ts:65]: I think we need to decalre it as async to have a better typing and ensure that await is called.
For instance, in camel.project.command.test.ts it is not. It is causing the title of the test to be incorrect `Should validate Create a Camel Project command GAV validation of [object Promise]`, see https://github.com/camel-tooling/camel-lsp-client-vscode/actions/runs/12173733325/job/33963279530?pr=1994#step:13:123

**Review by apupier — CHANGES_REQUESTED** [review]

File: src/test/suite/camel.project.command.test.ts [hunk 7]
```diff
@@ -20,12 +20,13 @@ import { expect } from 'chai';
 import { NewCamelQuarkusProjectCommand } from '../../commands/NewCamelQuarkusProjectCommand';
 import { NewCamelSpringBootProjectCommand } from '../../commands/NewCamelSpringBootProjectCommand';
 
-describe('Should validate Create a Camel Project command', function () {
+describe.only('Should validate Create a Camel Project command', function () {
```
Comment (apupier) [inline src/test/suite/camel.project.command.test.ts:23]: do not forget to remove the .only

**Review by apupier — APPROVED** [review]

**Review by djelinek — APPROVED** [review]

**Pull request merged**